### ISSUE TITLE
Add more tests for the API surface

### ISSUE THREAD

**Issue by domenic** [domenic/promises-unwrapping#68]
Currently only `then` and `all` have tests, really. (The former via the Promises/A+ tests.) Some of the other methods are tested implicitly by the `all` tests. But a deeper and more unit-test level coverage of things like `resolve`, `reject`, `cast`, and `race` would be good, and would help contribute to test-262 eventually.

**Comment by getify** [comment]
FWIW, that's precisely what I'm working on as part of my http://github.com/getify/native-promise-only polyfill.

**Comment by domenic** [comment]
Oooh excellent, will probably steal as they become available.

**Comment by getify** [comment]
Of course, I'm hoping we can collaborate, since I don't have nearly as much experience with such test case creation around promises as you all obviously do. :)

**Comment by getify** [comment]
Just FYI: some work is now in progress to create tests for the Promise.\* API surface.

https://github.com/getify/native-promise-only/pull/13

Would definitely encourage participation so we can make sure the test suite is as good and solid as possible. The goal is to have one central test-suite everyone can agree on and participate in, just like we have for the A+ part. :)

**Comment by stefanpenner** [comment]
+1
